src/Conference.tsx

```diff
@@ -418,100 +424,105 @@ const Conference: React.FC = (props) => {
     setSettingsModalOn(!settingsModalOn)
   }
 
-  const handleSettingsSubmit = async (selectedVideoDevice: MediaDeviceInfo | undefined, selectedAudioDevice: MediaDeviceInfo | undefined) => {
-    toggleSettings();
-    try {
-      setAudioAndVideoDevice(selectedAudioDevice, selectedVideoDevice);
-      const mediaDevice = await getAudioAndVideoDevice();
+  const handleSettingsSubmit = async (selectedVideoDevice: MediaDeviceInfo | undefined, selectedAudioDevice: MediaDeviceInfo | undefined, reloadApp: boolean) => {
+    if (reloadApp) {
```

Comment:
We have a little bit of a logic flaw here... If someone changes the devices and the TURN server config the device changes won't get persisted in local storage

Reply:
Perhaps something like this:

```
    toggleSettings();
    setAudioAndVideoDevice(selectedAudioDevice, selectedVideoDevice);

    if (reloadApp) {
      await bandwidthRtc.disconnect();
      window.location.reload();
    } else {
      try {
        const mediaDevice = await getAudioAndVideoDevice();
        try {
          if (localStream) {
            await bandwidthRtc.unpublish(localStream);
          }
        } finally {
          setLocalStream(undefined);
        }
        const publishResponse = await bandwidthRtc.publish(
            mediaDevice,
            undefined,
            'usermedia'
        );
        setLocalStream(publishResponse);
      } catch (e) {
        console.log("Error publishing... Skipping", e);
      }
    }
```